addons/mail/models/discuss/discuss_channel.py

```diff
@@ -446,15 +458,7 @@ def _action_unfollow(self, partner=None, guest=None):
             target = partner or guest
             target._bus_send_store(custom_store, notification_type="discuss.channel/leave")
             return
-        notification = Markup('<div class="o_mail_notification">%s</div>') % _(
-            "left the channel"
-        )
-        # sudo: mail.message - post as sudo since the user just unsubscribed from the channel
-        member.channel_id.sudo().message_post(
-            body=notification, subtype_xmlid="mail.mt_comment", author_id=partner.id
-        )
-        # send custom store after message_post to avoid is_pinned reset to True
-        member._bus_send_store(custom_store, notification_type="discuss.channel/leave")
+        self._post_leave_notification(partner.id, member)
```

Comment:
Always prefer working with recordset rather than id.
And don't need to give both partner and member as member itself contains more information already.